.github/workflows/docker-deploy.yml

```diff
@@ -31,9 +31,10 @@ jobs:
           context: ./
           file: ./Dockerfile
           builder: ${{ steps.buildx.outputs.name }}
+          platforms: linux/amd64,linux/arm64
           push: true
-          tags: ${{ env.DEPLOY_NAME }}
+          tags: ${{ steps.meta.outputs.tags }}
```

Comment:
Not sure I understand this change. What is the value of `steps.meta.output.tag`?

In my recent update to this file, I've made the value of `env.DEPLOY_NAME` conditional (it's set to `master` on a `push` event, and to e.g. `v5.3` for a `release` event. So this line will take some tweaking. I'm not sure if we can have the same tag for two different images (`arm` and `amd`)? If not, we should make sure `tags` is set to a sane value combing the `master`/version tag, and a tag indicating the plafform.

Reply:
I just looked and I did this wrong. 'steps.meta.outputs.tags' depends on a previous step which I didn't add, docker/metadata-action. If we add docker/metadata-action then the tags and labels are much handled automatically and can have many different values. I'm actually not sure if it is actually required or if it's worth it. This is [the documentation](https://github.com/docker/metadata-action#tags-input). Because I've never pushed an image to hub.docker.com, I'm still unsure how this will all work or what tags are needed.

I don't know how you feel about pushing to docker hub and seeing if it works. I don't know if there is a test docker hub to try this on. If it were me, I'd just push this to docker hub and figure out the right way, but Docker hasn't looked at my open source application yet. They sent an email basically saying they'll get to it eventually.

Reply:
For me it would be fine to test it on Dockerhub! Also looks like a good idea to use the `docker/metadata-action` action. Do you want to try to integrate that in this PR? If so, the goal would be to push versioned tags (e.g. `v5.2.3`) to Dockerhub on GitHub `release` events, and use the `master` tag when something is pushed to the GitHub `master` branch (which I now implemented without the `metadata-action`, see [here](https://github.com/gollum/gollum/commit/0c36cdf5ba080f0c5e8e925d1a446fda2fbcfc8b)). If that would cost you too much time, just let us know!